### PR TITLE
Update contributor guide and add doc for making local changes

### PR DESCRIPTION
**Pull request by rkuchan**
Hi everyone, I’m Regan and I’m interested in technical writing so Matt said I could help with OSPC’s docs. This is my first pull request so please give me your feedback.

I’ve done two main things in this branch. First, I edited the contributor_guide.rst. I added an introduction to the top of the file, altered some of the text, and added links to GitHub’s glossary so that new users can get reliable definitions of new terms.

Second, I added make_local_change.rst to the docs/source folder. This file documents how users can make a (simple) change to their local tax-calculator directory. It includes two screenshots (screen-shot-1.png and screen-shot-2.png) which are also stored in the docs/source folder. The contributor_guide now links to make_local_change.

Possible issues/ the reason for the WIP tag:

- “make html” worked on my machine (the images came out fine and the formatting was fine) but I am concerned that this will not continue to function well on taxcalc.readthedocs.org since the layout is a little different on rtd than on local html builds.
- I do not know how to make make_local_change show in the contents on taxcalc.readthedocs.org.
- I wasn’t sure where to store the two screenshots. They are currently in docs/source. 
- To link to make_local_change from the contributor_guide I used “taxcalc.readthedocs.org/en/latest/make_local_change.html’. I’m not sure if there is a better way to do this – the link does not work in my local html build, but I expect it to work on rtd.

Moving forward, I would like to add a discussion of how to handle .DS_Store on Macs, add helpful screenshots to the contributor_guide, and add more examples of making local changes to make_local_change.

### CODE REVIEW

File: docs/source/contributor_guide.rst [hunk 1]
```diff
@@ -117,14 +122,29 @@ For example usage, you can view our sample notebooks:
...
+.. _`Git setup`: https://help.github.com/articles/set-up-git/
+.. _`Fork`: https://help.github.com/articles/github-glossary/#fork
 .. _`password setup`: https://help.github.com/articles/caching-your-github-password-in-git/
-.. _`Tax Calculator's github page`: https://github.com/OpenSourcePolicyCenter/Tax-Calculator
+.. _`Tax Calculator's GitHub page`: https://github.com/OpenSourcePolicyCenter/Tax-Calculator
+.. _`repository`: https://help.github.com/articles/github-glossary/#repository
+.. _`push`: https://help.github.com/articles/github-glossary/#push
+.. _`pull`: https://help.github.com/articles/github-glossary/#pull
 .. _`Github Flow`: https://guides.github.com/introduction/flow/    
 .. _`10 Minues To TaxCalc`: http://nbviewer.ipython.org/github/OpenSourcePolicyCenter/Tax-Calculator/blob/master/docs/notebooks/10_Minutes_to_Taxcalc.ipynb
```
Comment (talumbau) [inline docs/source/contributor_guide.rst]: `Minues` -> `Minutes`

Comment (rkuchan) [inline docs/source/contributor_guide.rst]: Thanks. Just fixed.

**Comment by rkuchan** [comment]
In the latest commit I changed the links that relied on the RTD links; they no longer rely on the RTD links and they now work in my local html build.
 
I think my changes are ready to be merged. I will check all links and images as soon as it builds on RTD – again, everything works in my local html build, but I am wary that the images will not continue to function well on the RTD since the layout is a little different.

**Comment by MattHJensen** [comment]
Ok. Merging and building on rtd. @rkuchan, check them out in 5 min.

**Comment by jlyons871** [comment]
Concerning the differences between local builds and the live version, other than the formatting they should have the same functionality, as you can see now that your changes have been merged. If you will be frequently editing the docs, and would like to double check your changes before committing, it may be useful to create your own free account on https://readthedocs.org/, which you can point to your local Github repository for testing purposes. 

It looks like you started to discover more about using .rst files, although [this](http://sphinx-doc.org/rest.html) has some good examples of what you can do with them, and [this](https://raw.githubusercontent.com/OpenSourcePolicyCenter/Tax-Calculator/master/docs/source/index.rst) index file (shown as a raw file) is where the main structure is determined by RTD.  

Moving forward I would add an `images` directory inside of `docs/source` and possibly add short descriptions into the image names. The current setup is ok for now, but if there will be more screenshots added in the future, I think this additional organization will be useful.

**Comment by rkuchan** [comment]
Thank you! Those are helpful resources. I will definitely create an account on RTD and then open a new PR to deal with the images.

**Pull request merged**